model-engine/model_engine_server/domain/use_cases/llm_model_endpoint_use_cases.py

```diff
@@ -1222,21 +1245,35 @@ async def execute(
                 "inputs": request.prompt,
                 "parameters": {
                     "max_new_tokens": request.max_new_tokens,
+                    "repetition_penalty": request.repetition_penalty,
                 },
             }
             if request.stop_sequences is not None:
                 args["parameters"]["stop"] = request.stop_sequences
             if request.temperature > 0:
                 args["parameters"]["temperature"] = request.temperature
                 args["parameters"]["do_sample"] = True
+                if request.top_k == -1:  # tgi set to None to consider all tokens.
```

Comment:
Ah alternatively, you can keep these as `Optional` like you have, but default to `None`.